### DISCUSSION TITLE
What is 2FA

### DISCUSSION THREAD

**Discussion by THenryInGH** [community/community#73030] [GitHub Education]
### Hi everyone,

You must secure your GitHub account by enabling [2FA](https://github.com/settings/security) and complete your [billing information](https://github.com/settings/billing/payment_information). Faced this problem when apply for student pack, anyone know how to solve it, thanks!

**Comment by InvisiblePro** [comment]
2FA is two-factor-authentication. It is useful to make your account more secure.
For enabling 2FA follow this steps:
- Login on [GitHub](https://githu.com)
-  Now, follow this link, [GitHub settings](https://github.com/settings/profile).
- Go to `Password and Authentication` tab
![image](https://github.com/community/community/assets/92221630/cea9e16c-c69b-49d5-ae76-071b26e84755)
- Scroll down to get 2FA option
- Follow the steps mentioned there.
- Your 2FA will be enabled.

Have a happy journey ✨

**Comment by yutent** [comment]
2FA is a pile of shit

**Comment by dakshithadissanayaka** [comment]
To complete your GitHub student pack application, you need to enable two-factor authentication (2FA) and add your billing information. These steps are required to ensure the security of your GitHub account and to verify your student status.

Here's a step-by-step guide on how to enable 2FA and add your billing information:

**Enable Two-Factor Authentication (2FA)**

1. Go to your GitHub account settings: [https://docs.github.com/articles/about-your-profile](https://docs.github.com/articles/about-your-profile)
2. In the left sidebar, click on "Security"
3. Under "Two-factor authentication", click on "Set up two-factor authentication"
4. Choose your preferred 2FA method:
    * **Authenticator app:** Download an authenticator app like Google Authenticator or Authy on your mobile device. Scan the QR code displayed on the GitHub page with your authenticator app.
    * **SMS:** Enter your phone number and click on "Continue". You will receive a verification code via SMS. Enter the code to complete the setup.

**Add Billing Information**

1. Go to your GitHub account settings: [https://docs.github.com/articles/about-your-profile](https://docs.github.com/articles/about-your-profile)
2. In the left sidebar, click on "Billing & plans"
3. Under "Payment methods", click on "Add payment method"
4. Select your preferred payment method:
    * **Credit card:** Enter your credit card details and click on "Add credit card".
    * **PayPal:** Click on "Connect with PayPal" and follow the instructions to link your PayPal account.

Once you have completed these steps, you should be able to complete your GitHub student pack application. If you still encounter any issues, contact GitHub support for further assistance.

**Comment by mituldas1999** [comment]
To complete your GitHub Student Pack application, you need to enable Two-Factor Authentication (2FA) and add your billing information. Here's how to do it:

Enable Two-Factor Authentication (2FA)
Access Account Settings: Go to your GitHub account settings by visiting [GitHub profile settings](https://docs.github.com/articles/about-your-profile).

Security Settings: In the left sidebar of your account settings, click on "Security".

Setup 2FA: Under the "Two-factor authentication" section, select "Set up two-factor authentication".

Choose 2FA Method:

Authenticator App:
Download an authenticator app (like Google Authenticator or Authy) on your mobile device.
Scan the QR code displayed on the GitHub page with your authenticator app.
SMS:
Enter your phone number and click "Continue".
You will receive a verification code via SMS. Enter this code to complete the setup.
Add Billing Information
Access Billing Settings: Again, go to your GitHub account settings at the same [link](https://docs.github.com/articles/about-your-profile).

Billing & Plans: In the left sidebar, click on "Billing & plans".

Add Payment Method: Under "Payment methods", click on "Add payment method".

Select Payment Method:

Credit Card: Enter your credit card details and click on "Add credit card".
PayPal: Click on "Connect with PayPal" and follow the instructions to link your PayPal account.
Once you complete these steps, your GitHub Student Pack application should be good to go. If you encounter any issues, don't hesitate to contact GitHub support for assistance.

**Comment by mgrinzPlayer** [comment]
> Authenticator App:
> Download an authenticator app (like Google Authenticator or Authy) on your mobile device.

Or any other software which has TOTP feature, on any device: desktop or mobile.
This device can even be ALWAYS OFFLINE. The only unavoidable requirement is: this device must have correct date and time.

You can even write your own application: you need HMACSHA1Digest function and follow RFC6238 algorithm. I'm using self-written program.
